modules/features2d/src/gftt.cpp

```diff
@@ -106,14 +107,15 @@ class GFTTDetector_Impl CV_FINAL : public GFTTDetector
                 cvtColor( image, grayImage, COLOR_BGR2GRAY );
 
             goodFeaturesToTrack( grayImage, corners, nfeatures, qualityLevel, minDistance, _mask,
-                                blockSize, gradSize, useHarrisDetector, k );
+                                 cornersQuality, blockSize, gradSize, useHarrisDetector, k );
         }
 
         keypoints.resize(corners.size());
         std::vector<Point2f>::const_iterator corner_it = corners.begin();
         std::vector<KeyPoint>::iterator keypoint_it = keypoints.begin();
-        for( ; corner_it != corners.end() && keypoint_it != keypoints.end(); ++corner_it, ++keypoint_it )
-            *keypoint_it = KeyPoint( *corner_it, (float)blockSize );
+        std::vector<float>::iterator quality_it = cornersQuality.begin();
+        for( ; corner_it != corners.end() && keypoint_it != keypoints.end() && quality_it != cornersQuality.end(); ++corner_it, ++keypoint_it, ++quality_it )
+            *keypoint_it = KeyPoint( *corner_it, (float)blockSize, -1, *quality_it );
```

Comment:
This part of code looks strange (it was strange before this patch too).

What should we do with "left" uninitialized items when `keypoint_it != keypoints.end() && quality_it != cornersQuality.end()` is triggered instead of `corner_it != corners.end()`?

Perhaps it makes sense to put after this loop consistency check:

```
CV_Assert(corner_it == corners.end());
```

Reply:
Noted with thanks